vllm/distributed/kv_transfer/kv_connector/v1/nixl_connector.py

```diff
@@ -322,7 +371,25 @@ def __init__(self, vllm_config: VllmConfig, engine_id: str):
         self.tp_group = get_tp_group()
 
         # KV Caches and nixl tracking data.
-        self.kv_caches: dict[str, torch.Tensor] = {}
+        self.kv_buffer_device: str = vllm_config.kv_transfer_config.kv_buffer_device.strip().lower()
+        assert self.kv_buffer_device == current_platform.device_type, f"--{self.kv_buffer_device}, {current_platform.device_type}"
+        self.device_kv_caches: dict[str, torch.Tensor] = {}
+        self.device = None
+
+        # cpu kv buffer for xfer
+        # used when xPU memory can not be registered under nixl
+        self.host_xfer_buffers: dict[str, torch.Tensor] = {}
+        self.use_host_buffer = True if _NIXL_SUPPORTED_XPU_TYPE.support(self.kv_buffer_device) else False
```

Comment:
Is there any reason why we cannot allows the GPU to use the host buffer?

Also, it is confusing that the tpu uses `kv_buffer_device: "tpu"` when it is using the host buffer.

I think we should allow the user to specify `kv_buffer_device`, then have:


```python
_NIXL_SUPPORTED_XPU_TYPE = {
        "cuda": ["cuda", "cpu"],
        "tpu": ["cpu"]
}

if self.kv_buffer_device not in _NIXL_SUPPORTED_XPU_TYPE[current_platform.platform()]:
   raise
   
if self.kv_buffer_device == "cuda":
   self.nixl_memory_type = "VRAM"
else:
   assert self.kv_buffer_device == "cpu":
   self.nixl_memory_type = "DRAM"
```

Reply:
cuda: cpu is not supported yet.